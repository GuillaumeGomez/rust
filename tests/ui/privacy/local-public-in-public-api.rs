// This ui test checks that private types present in public APIs get
// a warning.

#![crate_type = "lib"]
#![deny(private_in_public)]

mod my_type {
    pub struct Field;
    pub struct MyType {
        pub field: Field,
    }

    impl MyType {
        pub fn builder() -> MyTypeBuilder {
                            //~^ ERROR
                            //~| WARN
            MyTypeBuilder { field: Field }
        }

        pub fn with_builder(_: MyTypeBuilder) {}
                               //~^ ERROR
                               //~| WARN
    }

    pub struct MyTypeBuilder {
        field: Field,
    }

    impl MyTypeBuilder {
        pub fn field(mut self, val: Field) -> Self {
            self.field = val;

            self
        }

        pub fn build(self) -> MyType {
            MyType { field: self.field }
        }
    }
}

pub use my_type::{MyType, Field};
