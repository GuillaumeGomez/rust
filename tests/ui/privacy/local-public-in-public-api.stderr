error: public but not reexported type `MyTypeBuilder` in public interface (error E0446)
  --> $DIR/local-public-in-public-api.rs:14:9
   |
LL |         pub fn builder() -> MyTypeBuilder {
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>
note: the lint level is defined here
  --> $DIR/local-public-in-public-api.rs:5:9
   |
LL | #![deny(private_in_public)]
   |         ^^^^^^^^^^^^^^^^^

error: public but not reexported type `MyTypeBuilder` in public interface (error E0446)
  --> $DIR/local-public-in-public-api.rs:20:9
   |
LL |         pub fn with_builder(_: MyTypeBuilder) {}
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>

error: aborting due to 2 previous errors

